src/com/owncloud/android/utils/ConnectivityUtils.java

```diff
@@ -44,4 +44,22 @@ public static boolean isAppConnected(Context context) {
         return cm.getActiveNetworkInfo() != null && cm.getActiveNetworkInfo().isConnected();
     }
 
+    public static boolean isNetworkActive(Context context) {
+
+        boolean isNetworkActive = true;
+
+        ConnectivityManager cm = (ConnectivityManager)context.
+                getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
+
+        // Check network availabality (no info object implies no connectivity)
+        if (activeNetwork == null || (activeNetwork!= null &&
```

Comment:
I would simplify to

```return (activeNetwork != null && activeNetwork.isConnectedOrConnecting())``` .

No ```isActiveNetwork``` variable declared nor assigned, easier condition.